src/app/(profile)/_actions/countProfileView.ts

```diff
@@ -1,14 +1,17 @@
+'use server'
 import {
   getProfileById,
   incrementProfileViewCountById,
 } from '@/backend/profile/profile.service'
 import { withSentry } from '@/utils/errHandling'
 
 export const countProfileView = withSentry(
-  async (profileId: string, userId: string) => {
+  async (profileId: string, userProfileId?: string) => {
```

Comment:
changed props names, and set userProfileId to be optional

Reply:
but you pass `session?.user.id` here which is not `userProfileId`. You need to change you `if` because it will not work.